### PR TITLE
fix Issue 21647 - pragma(msg) should be able to print a void type

### PR DESCRIPTION
**Pull request by WalterBright**
This:

1. changes printing of a void expression from `<void>` which is not valid D code to `cast(void)0`
2. brings `pragma(msg)` for PragmaDeclaration and PragmaStatement into harmony
3. adds an assert check for unsupported pointer sizes

### CODE REVIEW

**Comment by WalterBright** [comment]
Supercedes https://github.com/dlang/dmd/pull/7316/

**Comment by WalterBright** [comment]
Merged ice8255.d into fix21647.d as it's the same issue.

**Comment by WalterBright** [comment]
The Azure failure is another heisenbug:
```
Unable to locate build via Azure API. Please upload with the Codecov repository upload token to resolve issue.
```

File: src/dmd/hdrgen.d [hunk 1]
```diff
@@ -1864,10 +1864,17 @@ public:
...
                 buf.writestring(t.toChars());
                 buf.writeByte(')');
-                if (target.ptrsize == 8)
+                if (target.ptrsize == 4)
+                    goto case Tuns32;
+                else if (target.ptrsize == 8)
                     goto case Tuns64;
                 else
-                    goto case Tuns32;
+                    assert(0);
```
Comment (ibuclaw) [inline src/dmd/hdrgen.d:1872]: This will cause a regression that PR #10051 fixed.

**Review by ibuclaw — DISMISSED** [review]

File: src/dmd/hdrgen.d (hunk 1 above)
Comment (WalterBright) [reply src/dmd/hdrgen.d:1872]: I see, that was for the 16 bit code. I submit that 16 bit support should be explicit. I'll make it so.

Comment (ibuclaw) [reply src/dmd/hdrgen.d:1873]: Thanks, though I don't remember why I did not have Tuns16 here unlike in other places.  I'll have to rebuild that compiler configuration again soon.

**Review by ibuclaw — APPROVED** [review]

**Comment by WalterBright** [comment]
The heisenbug plague has returned:
```
->  Pinging Codecov
https://codecov.io/upload/v4?package=bash-20210129-7c25fce&token=secret&branch=pull%2F12211&commit=c867a41f72a6be4ba50a39d3739a3414d8fb8cb5&build=40309&build_url=&name=&tag=&slug=dlang%2Fdmd&service=circleci&flags=&pr=12211&job=0&cmd_args=p,Z
{'detail': ErrorDetail(string='Build has already finished, uploads rejected.', code='not_found')}
404
==> Uploading to Codecov
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 2329k  100    97  100 2329k    587  13.7M --:--:-- --:--:-- --:--:-- 13.8M
    {'detail': ErrorDetail(string='Build has already finished, uploads rejected.', code='not_found')}

Exited with code exit status 1
CircleCI received exit code 1
```

**Review by RazvanN7 — APPROVED** [review]

**Comment by MoonlightSentinel** [comment]
This introduced a regression: https://issues.dlang.org/show_bug.cgi?id=21740

**Pull request merged**